‘strncpy’ specified bound 128 equals destination size

#OS: Fedora 31
make check-TESTS command ends with this error.

src/write_splunk_test.c: In function ‘test_splunk_transforms’:
src/write_splunk_test.c:201:7: error: ‘strncpy’ specified bound 128 equals destination size [-Werror=stringop-truncation]
  201 |       strncpy(vl.plugin_instance, cases[i].plugin_instance,
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  202 |               sizeof(vl.plugin_instance));
      |               ~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/write_splunk_test.c:204:7: error: ‘strncpy’ specified bound 128 equals destination size [-Werror=stringop-truncation]
  204 |       strncpy(vl.type, cases[i].type, sizeof(vl.type));
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/write_splunk_test.c:206:7: error: ‘strncpy’ specified bound 128 equals destination size [-Werror=stringop-truncation]
  206 |       strncpy(vl.type_instance, cases[i].type_instance,
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  207 |               sizeof(vl.type_instance));
      |               ~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors

--- a/src/write_splunk_test.c
+++ b/src/write_splunk_test.c
@@ -197,14 +197,20 @@ DEF_TEST(splunk_transforms) {
     char dims_json[WS_METRIC_MAX_NAME_LEN] = {0};
 
     sstrncpy(vl.plugin, cases[i].plugin, sizeof(vl.plugin));
-    if (cases[i].plugin_instance != NULL)
+    if (cases[i].plugin_instance != NULL) {
       strncpy(vl.plugin_instance, cases[i].plugin_instance,
               sizeof(vl.plugin_instance));
-    if (cases[i].type != NULL)
+      vl.plugin_instance[sizeof(vl.plugin_instance) - 1] = '\0';
+    }
+    if (cases[i].type != NULL) {
       strncpy(vl.type, cases[i].type, sizeof(vl.type));
-    if (cases[i].type_instance != NULL)
+      vl.type[sizeof(vl.type) - 1] = '\0';
+    }
+    if (cases[i].type_instance != NULL) {
       strncpy(vl.type_instance, cases[i].type_instance,
               sizeof(vl.type_instance));
+      vl.type_instance[sizeof(vl.type_instance) - 1] = '\0';
+    }
 
     ws_transform(dims_json, metric_name, &vl, &pl_config);
     EXPECT_EQ_STR(cases[i].expected_name, metric_name);

